collector/src/main/java/com/bitmovin/analytics/features/FeatureFactory.kt

```diff
@@ -0,0 +1,5 @@
+package com.bitmovin.analytics.features
+
+interface FeatureFactory {
+    fun createFeatures(): Collection<Feature<*>>
+}
```

Comment:
I dont see a big value of this interface and its implementations, as adapters could have same method `createFeatures` (defined by base class) that would produce list of features. so in `PlayerAdapterBase` I would implement `registerFeatures` as abstract method, that returns features and has to be implemented